### PR TITLE
HPCC-20662 Spark xsd and xsl changes to modify labels in configmgr

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.garder@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@rpastrana Please take a quick look at the changes you requested.

File: initfiles/componentfiles/configxml/sparkThor.xsd [hunk 1]
```diff
@@ -32,7 +32,7 @@
                 </xs:appinfo>
             </xs:annotation>  
             <xs:sequence>
-                <xs:element name="Instance" maxOccurs="unbounded">
+                <xs:element name="SparkMasterInstance" maxOccurs="unbounded">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/sparkThor.xsd]: Limit this to a single instance. Also, are spaces allowed? If so, "Spark Master Instance" ?

File: initfiles/componentfiles/configxml/sparkThor.xsl.in [hunk 2]
```diff
@@ -38,7 +38,7 @@
         </xsl:call-template>
         <xsl:call-template name="printVariable">
             <xsl:with-param name="var" select="'SPARK_MASTER_HOST'"/>
-            <xsl:with-param name="val" select="Instance/@netAddress"/>
+            <xsl:with-param name="val" select="SparkMasterInstance/@netAddress"/>
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/sparkThor.xsl.in]: if it's possible to have multiple sparkMasterinstances, we might need to take the first one always.
something like select="SparkMasterInstance[1]/@netAddress"/ ?

**Review by rpastrana — CHANGES_REQUESTED** [review]
@Michael-Gardner couple of comments

File: initfiles/componentfiles/configxml/sparkThor.xsd (hunk 1 above)
Comment (Michael-Gardner) [reply initfiles/componentfiles/configxml/sparkThor.xsd]: Spaces aren't allowed.  Tried that but it's not valid unfortunately.

File: initfiles/componentfiles/configxml/sparkThor.xsl.in (hunk 2 above)
Comment (Michael-Gardner) [reply initfiles/componentfiles/configxml/sparkThor.xsl.in]: We should only ever have one master I think.  Perhaps I should try to make that a unique element?

Comment (rpastrana) [reply initfiles/componentfiles/configxml/sparkThor.xsl.in]: Yes, this should be a single element

**Comment by richardkchapman** [comment]
Needs rebasing

**Comment by Michael-Gardner** [comment]
@rpastrana Rebased and reverted the Instance element change that was causing an issue.

**Review by rpastrana — CHANGES_REQUESTED** [review]
@Michael-Gardner 
line 41 in the xsl.in file should ensure we're only using the first instance of Instance/@netaddress
Otherwise we might set multiple master nodes.
Also, very minor but the copyright date should be 2018

**Comment by Michael-Gardner** [comment]
Made the change that Rodrigo requested, and tested it.
@richardkchapman ready for merge.

**Pull request merged**